pkg/apis/serving/v1alpha1/service_validation.go

```diff
@@ -180,3 +181,27 @@ func (rt *ReleaseType) Validate(ctx context.Context) *apis.FieldError {
 
 	return errs.Also(rt.Configuration.Validate(ctx).ViaField("configuration"))
 }
+
+func (s *Service) validateCreatorAndModifier(ctx context.Context) *apis.FieldError {
+	if len(s.GetAnnotations()) == 0 {
+		return nil
+	}
+	var errs *apis.FieldError
+	if !apis.IsInUpdate(ctx) {
+		return nil
+	}
+	original := apis.GetBaseline(ctx).(*Service)
+	if original.GetAnnotations()[serving.CreatorAnnotation] != s.GetAnnotations()[serving.CreatorAnnotation] {
+		errs = errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("annotation %q is immutable", serving.CreatorAnnotation),
+			Paths:   []string{serving.CreatorAnnotation},
+		})
+	}
+	if equality.Semantic.DeepEqual(original.Spec, s.Spec) && original.GetAnnotations()[serving.UpdaterAnnotation] != s.GetAnnotations()[serving.UpdaterAnnotation] {
+		errs = errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("annotation %q is immutable", serving.UpdaterAnnotation),
```

Comment:
I think we want an invalid value error rather than an immutable error here. Updater is not immutable, but changing it when only metadata changes it is invalid. You should be able to use one of the defined FieldError types instead of defining inline.